ext/standard/var.c

```diff
@@ -652,9 +652,13 @@ PHP_FUNCTION(var_export)
 }
 /* }}} */
 
-static void php_var_serialize_intern(smart_str *buf, zval *struc, php_serialize_data_t var_hash);
+static void php_var_serialize_intern(smart_str *buf, zval *struc, php_serialize_data_t var_hash, bool unique, bool subtract_rc);
```

Comment:
It would be great to find better names for ``unique`` and ``subtract_rc``.
``unique`` may be replaced by ``may_be_in_dup_array`` or something simiar (negative meaning)
``subtract_rc`` may be replaced by ``dup_array``. It should be set to ``true`` in ``PHP_FUNCTION(serialize)`` checking IS_CV and refcount. If it's set we may ignore corresponding refcounter, otherwise we check if it's above 1.